src/coreclr/jit/gcencode.cpp

```diff
@@ -1578,6 +1578,15 @@ size_t GCInfo::gcInfoBlockHdrSave(
         assert(header->epilogCount <= 1);
     }
 #endif
+    if (compiler->UsesFunclets() && compiler->info.compFlags & CORINFO_FLG_SYNCH && compiler->opts.compDbgEnC)
+    {
+        // While the sync start offset and end offset are not used by the stackwalker/EH system
+        // in funclets mode, we do need to know if the code is synchronized if we are generating
+        // and edit and continue method, so that we can properly manage the stack during a Remap
+        // operation. Instead of inventing a new encoding, just encode some non-0 offsets into these fields.
+        header->syncStartOffset = 1;
+        header->syncEndOffset   = 2;
```

Comment:
Would it make sense to set `syncEndOffset` to `1` as well? That way the region becomes effectively empty and invalid. It would make it easier to detect this abuse of the encoding in diagnostic tools for both `FEATURE_EH_FUNCLETS` and legacy builds.

Reply:
That's a decent point. I don't think we have any diagnostic tools that care, but data that is patently invalid would be easier to check for. As I recall, I'd need to relax an assert or two, but that's no big deal.

Reply:
Done. Looks like it was just the one assert which was a problem.